Plugin refactor fixes

rtnpro opened the pull request:
Fixes for plugin refactor.

rtnpro committed:
Fixed typo in volume by category plugin.

rtnpro committed:
Refactored volume plugins to be stateless

on top of new refactored base plugins.

rtnpro committed:
Refactored releng base plugin to use refactored base plugins.

rtnpro committed:
Bugfixes in releng base plugin

along with updated tests.

rtnpro committed:
Use absolute imports in releng plugins.

rtnpro committed:
Add missing import in volume/simple plugin.

ralphbean commented on statscache_plugins/volume/by_package.py:
```diff
@@ -1,28 +1,29 @@
 import collections
 
 import statscache.plugins
+from statscache_plugins.volume.utils import VolumePluginMixin
```
I think you forgot to commit this `utils.py` file.  Take a look in your working dir -- are there any others?

rtnpro replied:
Oops! sorry. Pushed the missing `utils.py`

rtnpro committed:
Added missing volume/utils.py.

rtnpro commented:
@ralphbean Removed pdb

ralphbean commented:
Cool.  Looks good to me.  Want to merge it @rtnpro?

rtnpro commented:
@ralphbean , let's wait for some time. I am pushing some more changes :D

I will tell you when I think it's ready.

rtnpro committed:
Refactored layout data for releng plugin.

- Add default_options containing conf for status mapping, and logic for marking a log as old
- Allow overriding default options at component level.

rtnpro commented:
@ralphbean 

Ok, I have pushed my changes. Please have a look at https://github.com/fedora-infra/fedora-releng-dash/pull/22 and try things out.

Let me know what you think :)

rtnpro committed:
Update handle method and it's signature for all plugins.

rtnpro committed:
Handle None return type by releng sub plugins.

rtnpro committed:
Handle missing 'owner' key in message for artifacts releng plugin.

rtnpro committed:
Fix calls to plugins' handle method during their initialization.

rtnpro commented:
@ralphbean,

I have been testing the plugins by running `fedmsg-hub` in statscache for some time now. I fixed almost all the bugs (AFAIK) that I came across during testing. I think it's ready to be merged. Also, I think that the project is now mature enough to be hosted in Beta mode ;)

What do you say?

ralphbean commented on statscache_plugins/releng/plugins/amis.py:
```diff
@@ -20,20 +21,21 @@ def __init__(self, *args, **kwargs):
                 continue
             status = 'completed'
-            arch = message['msg']['image_name'].split('.')[1]
+            # arch = message['msg']['image_name'].split('.')[1]
```
Instead of commenting these out, just delete them?

ralphbean commented:
Looks good to me.  Let's work on packaging and staging deployment as discussed in IRC.
